armi/utils/pathTools.py

```diff
@@ -263,23 +263,30 @@ def cleanPath(path):
             "You tried to delete {0}, but it does not seem safe to do so.".format(path)
         )
 
-    for i in range(3):
-        try:
-            if os.path.exists(path) and os.path.isdir(path):
-                shutil.rmtree(path)
-            elif not os.path.isdir(path):
-                # it's just a file. Delete it.
-                os.remove(path)
-        except:
-            if i == 2:
-                pass
-            # in case the OS is behind or something
+    try:
+        if armi.MPI_RANK == 0:
+            clearPath(path)
+        while os.path.exists(path):
             sleep(0.1)
```

Comment:
This bare `sleep(0.1)` gives me pause.

I'm not sure these bare `sleep()` calls are 100% sufficient.  What if this loop spins for an hour? Surely that should be a FAIL, right?

Should we perhaps have a max amount of time the loop can spin, waiting for the deletion, before it fails? 10 minutes? What do you think?

Reply:
The method does need to be updated to use error handlers and explicitly catch exceptions to cover cases such as a file locked by another process... I'll add these.
Any sleep is undesirable, but might be good to have (along with logging) to account for situations such as a  "lag" between the OS response and actual deletion on the disk, which appears to be an issue on Windows.  My guess is at most a delay of 0.1 to 1 seconds would be acceptable to a user, before returning an error.

Reply:
> My guess is at most a delay of 0.1 to 1 seconds would be acceptable to a user, before returning an error.

1 second won't be enough. If the code is run on a large enough cluster there can be big IO delays if, says, 1000 cores are writing to the same disk at the same time. Perhaps 1 minute.